core/shared/src/main/scala/zio/logging/LogFilter.scala

```diff
@@ -284,10 +284,39 @@ object LogFilter {
     val mappingsSorted = mappings.map(splitNameByDotAndLevel.tupled).sorted(nameLevelOrdering)
     val nameGroup      = group.map(splitNameByDot)
 
+    @tailrec
+    def globStarCompare(l: List[String], m: List[String]): Boolean =
+      (l, m) match {
+        case (_, Nil)                       => true
+        case (Nil, _)                       => false
+        case (l @ (_ :: ls), m @ (mh :: _)) =>
+          // try a regular, routesCompare or check if skipping paths (globstar pattern) results in a matching path
+          l.startsWith(m) || compareRoutes(l, m) || globStarCompare(ls, m)
+      }
+
+    @tailrec
+    def compareRoutes(l: List[String], m: List[String]): Boolean =
+      (l, m) match {
+        case (_, Nil)                                  => true
+        case (Nil, _)                                  => false
+        case (_ :: ls, "*" :: ms)                      => true && compareRoutes(ls, ms)
+        case (l, "**" :: ms)                           => globStarCompare(l, ms)
+        case (lh :: ls, mh :: ms) if !mh.contains("*") =>
+          lh == mh && compareRoutes(ls, ms)
+        case (l @ (lh :: ls), m @ (mh :: ms))          =>
+          val (result, unprocessed) = mh
+            .split('*')
+            .toList
+            .foldLeft((true, lh)) { case ((matchResult, lr), p) =>
+              (matchResult && lr.containsSlice(p)) -> lr.drop(lr.indexOfSlice(p) + p.size)
```

Comment:
if `matchResult` is false `lr.drop(lr.indexOfSlice(p) + p.size` is unnecessary computation

Reply:
add a dedicated method to check it recursive and only use only indexOfSlice to look fo a match